integration/test_collection.py

```diff
@@ -97,6 +97,10 @@ def test_insert_with_no_generic(collection_factory: CollectionFactory) -> None:
     assert prop == "some name"
 
 
+def test_insert_with_refs(collection_factory: CollectionFactory) -> None:
```

Comment:
what is this for?

Reply:
Haha, I missed out adding the test. I will put it in now!